extensions/libraries/redcore/database/sqlparser/sqlparser.php

```diff
@@ -1160,79 +1373,103 @@ private function process_group($tokens, $select) {
 		return $out;
 	}
 
-	private function removeParenthesisFromStart($token) {
+	private function removeParenthesisFromStart($token)
+	{
 
 		$parenthesisRemoved = 0;
 
 		$trim = trim($token);
-		if ($trim !== "" && $trim[0] === "(") { // remove only one parenthesis pair now!
+		if ($trim !== "" && $trim[0] === "(")
+		{ // remove only one parenthesis pair now!
 			$parenthesisRemoved++;
 			$trim[0] = " ";
-			$trim = trim($trim);
+			$trim    = trim($trim);
 		}
 
 		$parenthesis = $parenthesisRemoved;
-		$i = 0;
-		$string = 0;
-		while ($i < strlen($trim)) {
+		$i           = 0;
+		$string      = 0;
+		while ($i < strlen($trim))
+		{
 
-			if ($trim[$i] === "\\") {
+			if ($trim[$i] === "\\")
+			{
 				$i += 2; // an escape character, the next character is irrelevant
 				continue;
 			}
 
-			if ($trim[$i] === "'" || $trim[$i] === '"') {
+			if ($trim[$i] === "'" || $trim[$i] === '"')
+			{
 				$string++;
 			}
 
-			if (($string % 2 === 0) && ($trim[$i] === "(")) {
+			if (($string % 2 === 0) && ($trim[$i] === "("))
+			{
 				$parenthesis++;
 			}
 
-			if (($string % 2 === 0) && ($trim[$i] === ")")) {
-				if ($parenthesis == $parenthesisRemoved) {
+			if (($string % 2 === 0) && ($trim[$i] === ")"))
+			{
+				if ($parenthesis == $parenthesisRemoved)
+				{
 					$trim[$i] = " ";
 					$parenthesisRemoved--;
 				}
 				$parenthesis--;
 			}
 			$i++;
 		}
+
 		return trim($trim);
 	}
 
-	private function initParseInfoExprList($parseInfo = false) {
-		if ($parseInfo === false) {
-			return array('processed' => false, 'expr' => "", 'key' => false, 'token' => false, 'tokenType' => "",
-						 'prevToken' => "", 'prevTokenType' => "", 'trim' => false, 'upper' => false);
+	private function initParseInfoExprList($parseInfo = false)
+	{
+		if ($parseInfo === false)
+		{
+			return array(
+				'processed' => false,
+				'expr' => "",
+				'key' => false,
+				'token' => false,
+				'tokenType' => "",
+				'prevToken' => "",
+				'prevTokenType' => "",
+				'trim' => false,
+				'upper' => false
+			);
 		}
 
-		$expr = $parseInfo['expr'];
+		$expr   = empty($parseInfo['expr']) ? array() : $parseInfo['expr'];
```

Comment:
@Kixo Here fix

Reply:
hmm... I havent checked but from the code wouldn't it be better if that empty check is placed above with that $parseInfo === false check, because it is returning array with proper false values

Reply:
In this case `$parseInfo` is return array, but `$parseInfo['expr']` is just empty string `''`. That would cause PHP error on next following line.

That's why I put check here for just $parseInfo['expr']